pkg/policy/api/l4.go

```diff
@@ -150,6 +150,19 @@ type Listener struct {
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:Required
 	Name string `json:"name"`
+
+	// Priority is used to select the Listener if multiple rules would apply different listeners
+	// to a policy map entry. Lower numbers indicate higher priority. If left out, a random, but
+	// unique priority on the range 10000-20000 is assigned, so that traffic will be
+	// consistently redirected to the same listener. If higher priority desired, a low unique
+	// number like 1, 2, or 3 should be explicitly specified here. If a lower than default
+	// priority is needed, then a unique number higher than 20000 should be explicitly
+	// specified. Numbers on the default range (10000-20000) are not allowed.
```

Comment:
The 10-20K behaviour to me sounds like it's implementation specifics leaking out into the API without firmly deciding about the target use cases and how users will use these priorities. In recent k8s upstream APIs, they've added priorities and in that case just added a range of 0-1000 ([link](https://network-policy-api.sigs.k8s.io/reference/spec/#policy.networking.k8s.io%2fv1alpha1.AdminNetworkPolicy)). I'm somewhat inclined to think that we don't need this level of complication expressed in the API.

If someone has listener policies today, then the ordering is arbitrary. If they upgrade and they want a particular listener to be used with priority vs another, then I think it's reasonable to expect them to define the priority for both listeners.

I'm also not quite sure what happens if someone starts to use a significant share of the 65K listeners, will that have bad bounding of the worst case behaviour? Limiting to 1K seems like it provides plentiful opportunities for use cases while reducing the scope for abuse. (I might argue even 100 is enough, easier to start with a smaller value and increase it later rather than start with a higher value and start to invalidate user policies later).